pulp_container/app/models.py

```diff
@@ -367,57 +367,52 @@ def generate_filename(instance, filename):
 
 class Upload(BaseModel):
     """
-    Model for tracking Blob uploads.
+    A model for tracking Blob uploads.
     """
 
     repository = models.ForeignKey(Repository, related_name="uploads", on_delete=models.CASCADE)
+    cumulative_size = models.BigIntegerField(default=0)
+
+
+class BlobTemporaryUpload(PulpTemporaryFile):
+    """
+    A model used for storing uploaded blob chunks in a temporary file.
+    """
 
+    upload = models.ForeignKey(Upload, related_name="blob_uploads", on_delete=models.CASCADE)
     offset = models.BigIntegerField(default=0)
 
-    file = models.FileField(
-        max_length=255,
-        null=True,
-        upload_to=generate_filename,
-        storage=FileSystemStorage(location=settings.CHUNKED_UPLOAD_DIR),
-    )
+    def save_chunk(self, chunk, chunk_size=None):
+        """Save the passed chunk to a temporary file and update offset for a current upload."""
+        self._init_temporary_file(chunk)
+        self._update_upload_size(chunk, chunk_size)
```

Comment:
Do those two methods not know `chunk` and `upload` from `self`?

Reply:
No. Not in the time of initialization.